library/ssl_msg.c

```diff
@@ -3745,6 +3745,45 @@ static int ssl_prepare_record_content(mbedtls_ssl_context *ssl,
                                            rec)) != 0) {
             MBEDTLS_SSL_DEBUG_RET(1, "ssl_decrypt_buf", ret);
 
+#if defined(MBEDTLS_SSL_EARLY_DATA) && defined(MBEDTLS_SSL_SRV_C)
```

Comment:
This is addressing #6360. It is better practice to address only one issue per PR (here #6331) and as the issue #6360 is scheduled for Q2 now, I'd prefer to move this to another PR and keep in this PR only the code addressing #6331.

Reply:
I will move this commit to other PR